### ISSUE TITLE
[explain in book] How to get from ResolvedDeclaration to ResolvedType (or vice versa)

### ISSUE THREAD

**Issue by daanschipper** [javaparser/javaparser#1500]
When trying to achieve the following I'm stumbling across the incompatibility between `ResolvedDeclaration` and `ResolvedType`. I want to find what  an `Expression` outputs when `toString()` is called upon that expression. 

First, I'm visiting all the `MethodDeclaration`s, checking if it's a toString method and analysing the return statement. I'm saving this information as pair `ResolvedMethodDeclaration : String` by calling `n.resolve()` on the `MethodDeclaration`. For instance `Foo : 'this is the toString of Foo'`.  

Next I visit the variables of interest, `Foo bar`, and want to get the toString definition of this variable. I'm resolving the variable with for instance `bar.calculateResolvedType()`, but this returns a `ResolvedType`. 

My question is how can I make the connection between `ResolvedDeclaration` and `ResolvedType`?

**Comment by ftomassetti** [comment]
I am not sure I understand what you are doing, please correct me if I misunderstood.

A ResolvedType can representing a bit of stuff: arrays, primitives, and declared type.
If and only if the type is a class, an interface or an enum then you could cast it to the appropriate subclass and get the corresponding declaration.

Are you looking for this?

```
myResolvedType.asReferenceType().getTypeDeclaration()
```

**Comment by daanschipper** [comment]
Yes, thanks for the explanation! That cleared some confusion on my part.

**Comment by ftomassetti** [comment]
Thank you for the question, it helps us understand what we should clarify!

**Comment by matozoid** [comment]
So should this go into the book or into the Javadoc? :-)

**Comment by ftomassetti** [comment]
I think it could be part of the book
